extensions/expanded-opcodes/README.md

```diff
@@ -0,0 +1,87 @@
+| `000000` | `ADD`           | `A ← A + B`                        | `ZNCV` |             |
+| `000001` | `ADC`           | `A ← A + B + C`                    | `ZNCV` |             |
+| `000010` | `SUB`           | `A ← A - B`                        | `ZNCV` |             |
+| `000011` | `SBB`           | `A ← A - B - C`                    | `ZNCV` |             |
+| `000100` | `RSUB`          | `A ← B - A`                        | `ZNCV` | (1)         |
+| `000101` | `RSBB`          | `A ← B - A - C`                    | `ZNCV` |             |
+| `000110` | `CMP`           | `_ ← A - B`                        | `ZNCV` |             |
+| `000111` | `RCMP`          | `_ ← B - A`                        | `ZNCV` |             |
+| `001000` | `OR`            | <code>A ← A &#124; B</code>        | `ZN`   | (5)         |
+| `001001` | `NOR`           | <code>A ← ~(A &#124; B)</code>     | `ZN`   | (5)         |
+| `001010` | `XOR`           | `A ← A ^ B`                        | `ZN`   | (5)         |
+| `001011` | `XNOR`          | `A ← ~(A ^ B)`                     | `ZN`   | (5)         |
+| `001100` | `AND`           | `A ← A & B`                        | `ZN`   | (5)         |
+| `001101` | `NAND`          | `A ← ~(A & B)`                     | `ZN`   | (5)         |
+| `001110` | `TEST`          | `_ ← A & B`                        | `ZN`   | (5)         |
+| `001111` |                 |                                    |        | reserved    |
+| `010000` | `SHL`           | `A ← A << B`                       | `ZN`   | (5)         |
+| `010001` | `ROL`           | `A ← A rotate left B`              | `ZN`   | (5)         |
+| `010010` | `LSHR`          | `A ← A >>> B`                      | `ZN`   | (5)         |
+| `010011` | `ASHR`          | `A ← A >> B`                       | `ZN`   | (5)         |
+| `010100` |                 |                                    |        | reserved    |
+| `010101` |                 |                                    |        | reserved    |
+| `010110` |                 |                                    |        | reserved    |
+| `010111` |                 |                                    |        | reserved    |
+| `011000` | `MOVZ`          | `A ← B`                            | None   | (7)         |
+| `011001` | `MOV` or `MOVS` | `A ← B`                            | None   | (8)         |
+| `011010` | `LOAD`          | `A ← MEM[B]`                       | None   |             |
+| `011011` | `STORE`         | `MEM[B] ← A`                       | None   | (2)         |
+| `011100` | `SLO`           | <code>A ← (A << 5) &#124; B</code> | None   | (3) (6)     |
+| `011101` |                 |                                    |        | reserved    |
+| `011110` | `READCR`        | `A ← CR[B]`                        | None   | (4) (6)     |
+| `011111` | `WRITECR`       | `CR[B] ← A`                        | None   | (2) (4) (6) |
```

Comment:
Why does this table have things which can already be encoded with a 1 byte opcode? There is no reason to have a strictly worse encoding for any of them.

Reply:
That's actually what prompted question number 1 in the PR description. Having them here as a mapping from the base-isa to larger opcodes doesn't make sense if there's no benefit which is why I suggested removing the expanded register extension and merging the larger registers into here. The 3 reserved bits in the PR would go to the registers and may or may not be shuffled around a bit. For the immediate mode, 2 of the bits would go to the immediate instead. Since we'd be removing the expanded registers extension, we'd be able to reclaim that bit as an additional `C` bit (or something else)

Reply:
To note: Having the extended registers prefix has the same byte length as using this extension for the base opcodes.